sporkfish/searcher/minimax.py

```diff
@@ -0,0 +1,253 @@
+import copy
+import logging
+import os
+import time
+from abc import ABC, abstractmethod
+from typing import Any, Callable, List, Optional, Tuple
+
+import chess
+import stopit
+from pathos.multiprocessing import ProcessPool
```

Comment:
[Minor] Ideally I'd like to clean up the imports (like this one should go in the lazy_smp file), but I think you might kill me for asking so much - should we wait till next PR? :)

Reply:
Cleaned up the imports a bit - do let me know if there is anything else regarding imports, happy to lump those in as part of this PR

Reply:
LGTM!